index.js

```diff
@@ -88,117 +82,81 @@ exports.init = function (sbot, config) {
-
-  hook(sbot.progress, function (fn) {
-    const prog = fn()
-    const p = ebt.progress()
-    if (p.target) prog.ebt = p
-    return prog
-  })
-
-  function onClose () {
-    // TODO: delete this, it seems to be used only in tests
-    sbot.emit('replicate:finish', ebt.state.clock)
+  // TODO: remove this when no one uses ssb-db anymore, because
+  // sbot.progress is defined in ssb-db but not in ssb-db2
+  if (sbot.progress) {
+    hook(sbot.progress, function (fn) {
+      const _progress = fn()
+      const ebtProg = ebt.progress()
+      if (ebtProg.target) _progress.ebt = ebtProg
+      return _progress
+    })
   }
 
   sbot.on('rpc:connect', function (rpc, isClient) {
     if (isClient) {
       const opts = { version: 3 }
-      const a = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
-      const b = rpc.ebt.replicate(opts, function (err) {
-        if (err) {
-          rpc.removeListener('closed', onClose)
-          rpc._emit('fallback:replicate', err)
+      const local = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
+      const remote = rpc.ebt.replicate(opts, (networkError) => {
+        if (networkError && getSeverity(networkError) >= 3) {
+          console.error(networkError)
         }
       })
-
-      pull(a, b, a)
-      rpc.on('closed', onClose)
+      pull(local, remote, local)
     }
   })
 
-  function block (from, to, blocking) {
-    if (isObject(from)) {
-      to = from.to
-      blocking = from.blocking
-      from = from.from
-    }
+  function request (destFeedId, requesting) {
+    if (!isFeed(destFeedId)) return
+    ebt.request(destFeedId, requesting)
+  }
+
+  function block (origFeedId, destFeedId, blocking) {
+    if (!isFeed(origFeedId)) return
+    if (!isFeed(destFeedId)) return
     if (blocking) {
-      ebt.block(from, to, true)
-    } else if (ebt.state.blocks[from] && ebt.state.blocks[from][to]) {
+      ebt.block(origFeedId, destFeedId, true)
+    } else if (
+      ebt.state.blocks[origFeedId] &&
+      ebt.state.blocks[origFeedId][destFeedId]
+    ) {
       // only update unblock if they were already blocked
-      ebt.block(from, to, false)
-    }
-    // if we blocked them, but happen to be connected, disconnect immediately.
-    if (blocking && sbot.id === from && ebt.state.peers[to]) {
-      // TODO: conn.disconnect might not work because we're passing a Feed ID
-      // while it expects a multiserver address
-      if (sbot.conn) sbot.conn.disconnect(to, function () {})
-      else if (sbot.gossip) sbot.gossip.disconnect(to, function () {})
+      ebt.block(origFeedId, destFeedId, false)
     }
   }
 
-  if (sbot.replicate.block) {
-    sbot.replicate.block.hook(function (fn, args) {
-      block.apply(this, args)
-      return fn.apply(this, args)
-    })
-  } else {
-    // remove this on next breaking change
-    Legacy(sbot, ebt)
-  }
-
-  function replicate(opts) {
+  function replicate (opts) {
     if (opts.version !== 2 && opts.version !== 3) {
```

Comment:
while you are changing things can you update this only check for version 3?

Reply:
Ok, done.

It's funny, though, that epidemic-broadcast-trees now doesn't utilize `version` in any way, it's just a number that is set to the stream `this` but never read. It would make more sense to put this check inside e-b-t.